### PR TITLE
Feat/278 step indicator

### LINKED ISSUE #278: Update sidebar step indicator
### Background 
To better align with CDT/CA State Design norms, we would like to update the design of the sidebar with the vital records application flow. We should do this before we expand the flow to include options for all 3 records types: birth, marriage and death. 

See [Figma screens](https://www.figma.com/design/CRJFJW74ZZ24mrEPS4YCmx/Vital-Records-Flow?node-id=2406-1087&p=f&t=GT1CPBY8CMTjP5iC-0) for design updates. 

Expected behavior is that the respective step in the sidebar will be filled when you are on that step: 

Image

### TODOs 

- [ ] Update sidebar design on all flow screens 
- [ ] Test functionality to ensure all edge cases and use cases are covered

### PR DESCRIPTION
**Pull request by machikoyasuda**
closes #278 

## What this PR does
- Changes the visual design of the step indicator: Uses this Vertical Tracker as the base for the HTML and CSS from the CA State Web Template https://template.webstandards.ca.gov/patterns/progress-tracker.html#Vertical-tracker
- Changes the logic of the indication bubbles: The current step and all completed steps are now blue.
- Indicator is hidden on mobile

## HTML/CSS
- Used https://template.webstandards.ca.gov/patterns/progress-tracker.html#Vertical-tracker for HTML and CSS. I removed the `article` element because there is no use for it in this component. 
- CSS: Used this Stack Overflow explanation of adjacent https://stackoverflow.com/questions/1817792/is-there-a-previous-sibling-selector https://tobiasahlin.com/blog/previous-sibling-css-has/ to select all preceeding siblings. 
- The HTML/CSS are written in a way that this code can be turned into a template that iterates over a list of steps with the URL path and step name. Aside from the last child/step needing the margin-bottom to be zero, every step has the same styles.

image

### CODE REVIEW

File: web/static/css/styles.css [hunk 1]
```diff
@@ -136,38 +136,28 @@ header {
   margin-bottom: 40px;
 }
 
-.cagov-step-list.sm {
-  padding-top: 8px;
+.vertical-tracker > li:has(~ .current) .dot-line-inner:before,
```
Comment (machikoyasuda) [inline web/static/css/styles.css:139]: Taking name suggestions for this component. Then I can rename the parent class (currently `vertical-tracker`) and the file name (currently `form-progress`) to the same uniform name!

Vertical tracker comes from here https://template.webstandards.ca.gov/patterns/progress-tracker.html#Vertical-tracker - though it differs in that the colored bubbles show you all your completed and current steps. Step indicator is from USWDS: https://designsystem.digital.gov/components/step-indicator/

**Comment by machikoyasuda** [comment]
Now ready for review

File: web/vital_records/templates/vital_records/_form_progress.html [hunk 2]
```diff
@@ -1,8 +1,56 @@
...
+<ul class="vertical-tracker list-unstyled sm m-t-0 d-none d-md-block d-lg-block">
+    <li class="row {% if url_name == 'request_name' %}current{% endif %}"
+        {% if url_name == 'request_name' %}aria-current="true"{% endif %}>
+        <div class="offset-md-2 col-md-10 pos-rel brd-md-left brd-gray-100 p-x-md">
+            <div class="timeline-dot top-0">
+                <span class="dot-line-inner bg-white brd-gray-100"></span>
+            </div>
+            <p class="small">Name</p>
+        </div>
+    </li>
```
Comment (angela-tran) [inline web/vital_records/templates/vital_records/_form_progress.html:10]: Would it make sense to extract this out as an include template?

Comment (machikoyasuda) [reply web/vital_records/templates/vital_records/_form_progress.html:10]: Yeah, I think it does. But I was thinking that would be in a future ticket. 

Once we add the marriage steps, we'll have to figure out how to represent each record type's steps (URLs and step name) and iterate the template over each step. 

Could be part of tomorrow's discussions.

File: web/static/css/styles.css (hunk 1 above)
Comment (angela-tran) [reply web/static/css/styles.css:139]: Thanks for the links explaining the usage of `:has` and `~` to select the previous siblings!
